pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -148,66 +152,162 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * Windows only. Read settings values from registry using schemaObj as a template.
  * @param schemaObj the object used as a template for navigating registry.
  * @param regKey the registry key obtained from nativeReg.openKey().
+ * @param pathParts the path to the current registry key
  * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, pathParts: string[]): RecursivePartial<settings.Settings>|null {
   let newObject: RecursivePartial<settings.Settings>|null = null;
 
   const schemaKeys = Object.keys(schemaObj);
+  const schemaKeysLowerCase = schemaKeys.map(k => k.toLowerCase());
+  const commonKeys: string[] = [];
+  const unknownKeys: string[] = [];
+  const specialObjectKeys: string[] = [];
+  let regValue: any;
+
   // ignore case
-  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
-  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
+  nativeReg.enumKeyNames(regKey).forEach((k) => {
+    if (k.toLowerCase() === 'integrations' && pathParts.length === 3 && pathParts[2] === 'WSL') {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'extensions' && pathParts.length === 2) {
+      specialObjectKeys.push(k);
+    } else if (k.toLowerCase() === 'mutedchecks' && pathParts.length === 3 && pathParts[2] === 'diagnostics') {
+      specialObjectKeys.push(k);
+    } else if (schemaKeysLowerCase.includes(k.toLowerCase())) {
+      commonKeys.push(k);
+    } else {
+      unknownKeys.push(k);
+    }
+  });
+  if (unknownKeys.length) {
+    console.log(`Unrecognized keys in registry at \\${ pathParts.join('\\') }: [${ unknownKeys.join(', ') }]`);
+  }
 
+  // First process the nested keys, then process any values
   for (const k of commonKeys) {
     const schemaVal = schemaObj[k];
```

Comment:
At this point, `schemaVal` may be a non-object (that is, a string or a number). Is that expected?

Reply:
I don't see how that can happen. `k` must name a registry key, which by definition will return an object. But it's an error if `k` on the schema side is a non-object (including non-array)

Reply:
As a user, I can create `HKCU\Defaults\debug` as a key (and put random values in it); up to this point we only checked that `schemaObj` has a key that matches, not that `schemaObj[k]` is an object.  It's an error case, but it's not an error case we're handling.

Reply:
I've added unit tests in `deploymentProfiles.spec.ts` that insert all kinds of incorrect data into the registry. It doesn't check the exact error messages, just verifies that invalid data doesn't end up in the loaded settings object.